### ISSUE TITLE
%%magic function breaks syntax coloring for HTML export

### ISSUE THREAD

**Issue by profedrini** [jupyter/nbconvert#105]
Within notebook:
![snap0000](https://cloud.githubusercontent.com/assets/8213196/9944466/2291c310-5d4d-11e5-88cc-be07b43ff74c.png)

but after "file |  Download as ..  | HTML":
![snap0001](https://cloud.githubusercontent.com/assets/8213196/9944477/2e00d696-5d4d-11e5-8770-2a1359016cba.png)

However, cells without %%magic will correctly export HTML colored:
![snap0002](https://cloud.githubusercontent.com/assets/8213196/9944507/56cdb468-5d4d-11e5-94b0-70e6d0871a6f.png)

**Comment by captainsafia** [comment]
Hm. I dug into this and it looks to be an issue with the `IPython3Lexer` in IPython. The lexer doesn't properly tag the remaining tokens if the string begins with a magic function so I assume this should be fixed in ipython.

```
>>> from IPython.lib import lexers
>>> lexer = lexers.IPython3Lexer()                                                                                                                                                               
>>> fragment = "%%time\nfor i in range(5):\nprint(n)"
>>> other_fragment = "for i in range(5):\nprint(n)"
>>> list(lexer.get_tokens(fragment))
[(Token.Operator, '%%'), (Token.Keyword, 'time'), (Token.Text, '\nfor i in range(5):\nprint(n)\n')]
>>> list(lexer.get_tokens(other_fragment))                                                                                                                                                       
[(Token.Keyword, 'for'), (Token.Text, ' '), (Token.Name, 'i'), (Token.Text, ' '), (Token.Operator.Word, 'in'), (Token.Text, ' '), (Token.Name.Builtin, 'range'), (Token.Punctuation, '('), (Token.Literal.Number.Integer, '5'), (Token.Punctuation, ')'), (Token.Punctuation, ':'), (Token.Text, '\n'), (Token.Name.Builtin, 'print'), (Token.Punctuation, '('), (Token.Name, 'n'), (Token.Punctuation, ')'), (Token.Text, '\n')]
```

**Comment by michaelpacer** [comment]
Changing this to no action, because if it is dependent on a change in the IPython `IPython3Lexer` it shouldn't affect our milestones here.

**Comment by t-makaro** [comment]
Closing as fixed by [IPython #11137](https://github.com/ipython/ipython/pull/11137)
